xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/objects/classes/DBListClass.java

```diff
@@ -282,8 +285,14 @@ public String getQuery(XWikiContext context)
                     select.append("doc." + idField);
                 } else {
                     select.append("idprop.value");
-                    fromStatements.add("StringProperty as idprop");
                     whereStatements.add("obj.id=idprop.id.id and idprop.id.name='" + idField + "'");
+                    // Get the from statements according to the type of property
+                    try {
+                        String classType = getPropertyType(classname, idField,context);
+                        fromStatements.add(classType + " as idprop");
+                    } catch (Exception e) {
+                        LOGGER.error("Failed to parse SQL script [{}]. Continuing with non-rendered script.", sql, e);
```

Comment:
I don't understand what this error log message has to do with the code executed in the ``try`` block above. Where are you "parsing the SQL script"? Also, I see that ``getPropertyType`` has its own try/catch block so what exceptions are you hoping to catch here?